Data/HashMap/Internal/Array.hs

```diff
@@ -442,9 +460,10 @@ map f = \ ary ->
     in run $ do
         mary <- new_ n
         go ary mary 0 n
+        return mary
   where
     go ary mary i n
-        | i >= n    = return mary
+        | i >= n    = return ()
```

Comment:
I have extracted these loop changes to https://github.com/haskell-unordered-containers/unordered-containers/pull/377.